zhaquirks/develco/air_quality.py

```diff
@@ -1,241 +1,90 @@
-        # input_clusters=[0, 1, 3, 32, 1026, 1029, 64515] output_clusters=[3, 10, 25]>
-        MODELS_INFO: [
-            (DEVELCO, "AQSZB-110"),
-            ("frient A/S", "AQSZB-110"),
-        ],
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: 0xC0C9,
-                DEVICE_TYPE: 1,
-                INPUT_CLUSTERS: [
-                    Identify.cluster_id,
-                    Scenes.cluster_id,
-                    OnOff.cluster_id,
-                ],
-                OUTPUT_CLUSTERS: [],
-            },
-            38: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: zha.DeviceType.TEMPERATURE_SENSOR,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    PowerConfiguration.cluster_id,
-                    Identify.cluster_id,
-                    PollControl.cluster_id,
-                    TemperatureMeasurement.cluster_id,
-                    RelativeHumidity.cluster_id,
-                    0xFC03,
-                ],
-                OUTPUT_CLUSTERS: [Identify.cluster_id, Time.cluster_id, Ota.cluster_id],
-            },
-        },
-    }
-
-    replacement = {
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: 0xC0C9,
-                DEVICE_TYPE: 1,
-                INPUT_CLUSTERS: [
-                    Identify.cluster_id,
-                    Scenes.cluster_id,
-                    OnOff.cluster_id,
-                ],
-                OUTPUT_CLUSTERS: [],
-            },
-            38: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: zha.DeviceType.TEMPERATURE_SENSOR,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    DevelcoPowerConfiguration,
-                    Identify.cluster_id,
-                    PollControl.cluster_id,
-                    DevelcoTemperatureMeasurement,
-                    DevelcoRelativeHumidity,
-                    DevelcoVOCMeasurement,
-                    EmulatedVOCMeasurement,
-                ],
-                OUTPUT_CLUSTERS: [Identify.cluster_id, Time.cluster_id, Ota.cluster_id],
-            },
-        },
-    }
+        cluster_revision: Final = ZCL_CLUSTER_REVISION_ATTR
+        reporting_status: Final = ZCL_REPORTING_STATUS_ATTR
+
+
+(
+    QuirkBuilder("frient A/S", "AQSZB-110")
+    .applies_to("Develco Products A/S", "AQSZB-110")
+    .replaces(DevelcoVOCMeasurement, endpoint_id=38)
+    .replaces(DevelcoPowerConfiguration, endpoint_id=38)
+    .sensor(
+        attribute_name=DevelcoVOCMeasurement.AttributeDefs.measured_value.name,
+        cluster_id=DevelcoVOCMeasurement.cluster_id,
+        endpoint_id=38,
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
+        state_class=SensorStateClass.MEASUREMENT,
+        unit=CONCENTRATION_PARTS_PER_BILLION,
+        fallback_name="VOC Level",
+        unique_id_suffix="voc_level",
```

Comment:
I'm aware the fallback name isn't used in HA and no translation key is specified (meaning we use the device class name: "Volatile organic compounds"), but maybe we should still rename that to "VOC level" to match HA conventions.
```suggestion
        fallback_name="VOC level",
        unique_id_suffix="voc_level",
```
The IKEA VINDSTYRKA quirk uses "VOC index" and "voc_index", but it's intentionally using `SensorDeviceClass.AQI`, as its measurement has no unit. We could still use "index" in the name here, too. Doesn't really matter.

Reply:
Sure! I think we should invest in custom pylint code to enforce these rules via CI.

Reply:
Or maybe just have zigpy be strict with it and throw an error?

Reply:
I think having zigpy throw an error might be too strict, since there are some times where we don't want to enforce this (due to abbreviations or special/custom names).
We could also just add a test that goes through all `EntityMetadata` and the `ChangedEntityMetadata` and verifies the `fallback_name`s, but it would still easily allow us to opt-out certain names. I'll get back to this soon.

Reply:
Right. I *think* we can do it with https://pylint.pycqa.org/en/latest/development_guide/how_tos/custom_checkers.html, allowing quirks in this repo to opt out on a case-by-case basis with a narrow `# noqa` directive.